sys/net/gnrc/link_layer/lorawan/gnrc_lorawan_mcps.c

```diff
@@ -358,6 +378,15 @@ void gnrc_lorawan_event_no_rx(gnrc_lorawan_t *mac)
         return;
     }
 
+    if (mac->mlme.adr_ack_cnt  > (CONFIG_LORAMAC_DEFAULT_ADR_ACK_LIMIT +
+         CONFIG_LORAMAC_DEFAULT_ADR_ACK_DELAY)) {
+             if (mac->last_dr) {
+                DEBUG("gnrc_lorawan_mcps: ADRACKReq: Decrement DR\n");
+                mac->last_dr--;
+                mac->mlme.adr_ack_cnt = CONFIG_LORAMAC_DEFAULT_ADR_ACK_LIMIT;
+             }
+    }
```

Comment:
This should work, but we shouldn't reset the counter to `CONFIG_LORAMAC_DEFAULT_ADR_ACK_LIMIT`.
The counter should keep increasing.

You can use what was proposed in This was already proposed in https://github.com/RIOT-OS/RIOT/pull/15995#discussion_r583743697

We should enforce these CONFIGs to be power of 2. Then, the code can be written with bit operations.

You could define conditions that you should reuse in other parts of the code and reflect the standard with those.

Reply:
Yes. This is true :)